src/Illuminate/Database/Query/Builder.php

```diff
@@ -2257,6 +2255,13 @@ public function getCountForPagination($columns = ['*'])
      */
     protected function runPaginationCountQuery($columns = ['*'])
     {
+        if ($this->groups || $this->havings) {
+            return [(object) ['aggregate' => $this->newQuery()
+                        ->from(new Expression('('.$this->toSql().') as '.$this->grammar->wrap('aggregate_table')))
+                        ->mergeBindings($this)
+                        ->count(['*']), ]];
```

Comment:
Would it be faster to count the ID if there's one? instead of counting all columns?

Reply:
We don't always know which columns are actually available. There is also a difference between `count(*)` and `count(id)`: https://stackoverflow.com/a/3003533/4848587

Reply:
the $columns param is being passed by not used, maybe it should be 
`->count($columns), ]];`